test/utils/session-core.test.js

```diff
@@ -0,0 +1,282 @@
+import { getDocuments } from '../../src/utils/network/session-core';
+
+const mockPodUrl = 'https://pod.example.com/';
+const mockSolidIdentityProvider = 'https://example.com/';
+let session = {};
+
+describe('setDocAclPermission', () => {
+  const setDocAclForUser = vi.fn(() => Promise.resolve());
+
+  beforeEach(() => {
+    session = {
+      fetch: vi.fn(),
+      info: {
+        webId: `${mockPodUrl}profile/card#me`
+      }
+    };
+  });
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('prints out correct documentUrl and webId for other user and calls setDocAclForUser', async () => {
+    const otherPodUsername = 'pod2';
+    const permissions = { read: true };
+
+    const documentUrl = 'https://pod.example.com/Drivers%20License/';
+    const webId = `https://${otherPodUsername}.${
+      mockSolidIdentityProvider.split('/')[2]
+    }/profile/card#me`;
+
+    await setDocAclForUser(session, documentUrl, 'update', webId, permissions);
+
+    expect(setDocAclForUser).toBeCalledWith(session, documentUrl, 'update', webId, permissions);
+    expect(documentUrl).toBe('https://pod.example.com/Drivers%20License/');
+    expect(webId).toBe('https://pod2.example.com/profile/card#me');
+  });
+});
+
+describe('setDocContainerAclPermission', () => {
+  const setDocAclForUser = vi.fn(() => Promise.resolve());
+
+  beforeEach(() => {
+    session = {
+      fetch: vi.fn(),
+      info: {
+        webId: `${mockPodUrl}profile/card#me`
+      }
+    };
+  });
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('calls setDocAclForUser 4 times', async () => {
+    const otherPodUsername = 'pod2';
+    const permissions = { read: true, write: true, append: true, control: false };
+    const containerUrl = 'https://pod.example.com/Documents/';
+    const urlsToSet = [
+      containerUrl,
+      `${containerUrl}Bank%20Statement/`,
+      `${containerUrl}Passport/`,
+      `${containerUrl}Drivers%20License/`
+    ];
+
+    const webId = `https://${otherPodUsername}.${
+      mockSolidIdentityProvider.split('/')[2]
+    }/profile/card#me`;
+
+    urlsToSet.forEach(async (url) => {
+      await setDocAclForUser(session, url, 'update', webId, permissions);
+    });
+
+    expect(setDocAclForUser).toBeCalledTimes(4);
+  });
+});
+
+describe('checkContainerPermission', () => {
+  const otherPodUsername = 'pod2';
+  const documentsContainerUrl = `https://${otherPodUsername}.${
+    mockSolidIdentityProvider.split('/')[2]
+  }/Documents/`;
+  let catchExecuted = false;
+  const mockGetSolidDataset = vi.fn((url) => Promise.resolve(mockSolidDatasetFrom(url)));
+  const mockGetSolidDatasetRejected = vi.fn(() => Promise.reject(Error('dataset not found')));
+
+  beforeEach(() => {
+    session = {
+      fetch: vi.fn(),
+      info: {
+        webId: `${mockPodUrl}profile/card#me`
+      }
+    };
+  });
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('expect try block to be ran if getSolidDataset resolves', async () => {
```

Comment:
This one also doesn't test anything outside the test. It just runs a function you defined on line 91.

Reply:
Removing test for this.